src/analyses/base.ml

```diff
@@ -1261,9 +1261,29 @@ struct
   *)
   (* TODO: deduplicate https://github.com/goblint/analyzer/pull/1297#discussion_r1477804502 *)
   let rec exp_may_signed_overflow ctx exp =
+    let binop = (GobOption.map2 Z.div )in
     let res = match Cilfacade.get_ikind_exp exp with
       | exception _ -> BoolDomain.MayBool.top ()
       | ik ->
+        let checkDiv e1 e2 =
+          match ctx.ask (EvalInt e1), ctx.ask (EvalInt e2) with
+          | `Bot, _ -> false
+          | _, `Bot -> false
+          | `Lifted i1, `Lifted i2 ->
+            ( let divisor_contains_zero = (ID.meet i2 (ID.of_int ik Z.zero) = ID.of_int ik Z.zero)  in
```

Comment:
This looks like it should work, but might be fragile because of some domain weirdness (refinements not taking place etc).
I think the safer check is to check whether `ID.is_bot` holds for the meet.

Reply:
... helps when you know the API. Thank you